pkg/clients/elasticache/elasticache.go

```diff
@@ -677,6 +679,16 @@ func GenerateCreateCacheClusterInput(p cachev1alpha1.CacheClusterParameters, id
 		SnapshotWindow:             p.SnapshotWindow,
 	}
 
+	if p.EngineVersion != nil {
+		version, err := getVersion(p.EngineVersion)
+		if err != nil {
+			return nil, err
+		}
+		c.EngineVersion = version
+	} else {
+		c.EngineVersion = p.EngineVersion
```

Comment:
This else clause is not needed if you keep
```
EngineVersion:              p.EngineVersion,
```

Reply:
yes but i think its easier to see that we doing some version stuff with this variable - i think we should leave it as is